src/structures/node.jl

```diff
@@ -56,10 +55,117 @@ struct CyclicRepresentative <: Cyclic end
 
 `StorageBehavior` in which the the cyclic behaviour is achieved within a strategic period.
 This implies that the initial level in individual representative periods can be different
-when using `RepresentativePeriods`
+when using `RepresentativePeriods`.
 """
 struct CyclicStrategic <: Cyclic end
 
+
+"""
+`AbstractStorageParameters` as supertype for individual parameters for `Storage` nodes.
+
+Storage parameters are used to provide the user the flexibility to include or not include
+capacities and variable and fixed OPEX parameters for charging, the storage level, and
+discharging.
+"""
+abstract type AbstractStorageParameters end
+
+"""
+    StorCapOpex <: AbstractStorageParameters
+
+A storage parameter type for including a capacity as well as variable and fixed operational
+expenditures.
+
+# Fields
+- **`capacity::TimeProfile`** is the is the installed capacity.
+- **`opex_var::TimeProfile`** is the variational operational costs per energy unit.
+- **`opex_fixed::TimeProfile`** is the fixed operational costs.
+"""
+struct StorCapOpex <: AbstractStorageParameters
+    capacity::TimeProfile
+    opex_var::TimeProfile
+    opex_fixed::TimeProfile
+end
+
+"""
+    StorCap <: AbstractStorageParameters
+
+A storage parameter type for including only a capacity. This implies that neither the usage
+of the `Storage`, nor the installed capacity have a direct impact on the objective function.
+
+# Fields
+- **`capacity::TimeProfile`** is the is the installed capacity.
+"""
+struct StorCap <: AbstractStorageParameters
+    capacity::TimeProfile
+end
+
+"""
+    StorCap <: AbstractStorageParameters
+
+A storage parameter type for including a capacity and variable operational expenditures.
+This implies that the installed capacity has no direct impact on the objective function.
+
+# Fields
+- **`capacity::TimeProfile`** is the is the installed capacity.
+- **`opex_var::TimeProfile`** is the variational operational costs per energy unit.
+"""
+struct StorCapOpexVar <: AbstractStorageParameters
+    capacity::TimeProfile
+    opex_var::TimeProfile
+end
+
+"""
+    StorCapOpexFixed <: AbstractStorageParameters
+
+A storage parameter type for including a capacity and fixed operational expenditures.
+This implies that the installed capacity has no direct impact on the objective function.
+
+# Fields
+- **`capacity::TimeProfile`** is the is the installed capacity.
```

Comment:
```suggestion
- **`capacity::TimeProfile`** is the installed capacity.
```

Reply:
The correction is included in a new commit, including all other suggested changes, as I thought it is beneficial to apply all of them as one.